posts/2019-02-08-googleemail2_deprecation.md

```diff
@@ -0,0 +1,122 @@
+---
+title: GoogleEmail2 Deprecation
+tags: haskell, yesod
+---
+
+I maintain an Auth [plugin][oauth2-google] for authenticating with Google OAuth2
+in a Yesod application. This plugin has always had functionality overlap with
+the [`GoogleEmail2`][google-email2] plugin in the `yesod-auth` package. Our
+Google plugin was present, removed (due to said overlap), then returned again,
+along with [some discussion][google-pr] about deprecating `GoogleEmail2` in
+favor of it. What was missing was the documentation for migrating.
+
+[oauth2-google]: http://hackage.haskell.org/package/yesod-auth-oauth2-0.6.1.0/docs/Yesod-Auth-OAuth2-Google.html
+[google-email2]: http://hackage.haskell.org/package/yesod-auth-1.6.5/docs/Yesod-Auth-GoogleEmail2.html
+[google-pr]: https://github.com/thoughtbot/yesod-auth-oauth2/pull/32#issuecomment-110013684
+
+Things lived happily in this state for some time, until the deprecation of the
+Google+ API sparked the discussion again. `GooglEmail2` relies on this API, but
+`OAuth2.Google` does not. Therefore, it makes more sense to push the ecosystem
+towards our plugin. That means we need to document the migration path, which is
+what this blog post is.
+
+## Caveat
+
+The following describes the fastest way to migrate your codebase, by changing as
+little about your application as is required to maintain existing functionality
+under the new plugin. However, I would consider it an introduction of Technical
+Debt. I encourage you to spend the time to actually alter your application to
+align better with how the new plugin does things. How to do that would be
+application-specific, so I don't offer concrete guidance here -- but my hope is
+that after following the "fast way" below, you will understand enough about the
+differences between the plugins to know what to best do in your own codebase.
+
+## Migration
+
+Actually changing plugins is as simple as you might expect:
+
+```diff
+-import Yesod.Auth.GoogleEmail2
++import Yesod.Auth.OAuth2.Google
+
+-authPlugins = [authGoogleEmailSaveToken clientId clientSecret]
++authPlugins = [oauth2GoogleScoped ["email", "profile"] clientId clientSecret]
```

Comment:
IIUC, this is currently giving the Google ID as the identifier instead of the email, which needs to be worked around in order to have compatibility with GoogleEmail2. Is there any straightforward way to get the old identifier behavior?